### PR TITLE
Hard fork proposal to address the Ice age

### PR DESCRIPTION
**Pull request by MadeofTin**
Included is a proposal for a hard fork that address the ice age, as well as lays out a commitment from the core developers to solve the problems with the ice age in a permanent way.

### CODE REVIEW

File: EIPS/eip-ice-age-fork.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+ - Update the mechanism so that behavior is predictable.
+ - Remove the Iceage entirely
+
+## Specification
+
+- Codename: Glacier Mountain
+
+### Activation
+  - `Block >= 9,199,999` on the Ethereum mainnet
```
Comment (shemnon) [inline EIPS/eip-ice-age-fork.md]: Why not 9,200,000?  The new calculation will be done on that block and not the old value with 30 second blocks.  And it keeps the triple zero endings.

File: EIPS/eip-ice-age-fork.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+The primary problem with the Ice Age is that it is included in the complex mechanism that targets block times, which is an entirely separate in purpose. What is worse is due to being intwined with that algorithm, it is very difficult to simulate or predict its effect on the network. To predict the impact of the ice age, you must both make assumptions about the difficulty of main-net in the future, and predict the effect of changes in difficulty to the impact on the ice age and thus block-times.
+
+This fork will push back the Iceage as far as possible and will give us time to update the Iceage to no longer have these design problems. There are two solutions to consider with-in that time frame.
+
+ - Update the mechanism so that behavior is predictable.
+ - Remove the Iceage entirely
+
+## Specification
+
+- Codename: Glacier Mountain
```
Comment (shemnon) [inline EIPS/eip-ice-age-fork.md]: Word order is Different from the title on line 3 and 14.  Either is fine, pick one.

File: EIPS/eip-ice-age-fork.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+discussions-to: https://github.com/ethereum/EIPs/pull/2387
+type: Meta
+status: Draft
+created: 2019/11/22
+requires: 1679
+---
+
+## Abstract
+
+This meta-EIP specifies the changes included in the Ethereum hard fork named Mountain Glacier. This hard fork addresses the impending Ice Age on Ethereum Main-net and includes a commitment to solving the problems with the ice age more permanently.
```
Comment (shemnon) [inline EIPS/eip-ice-age-fork.md]: "Mainnet" is more common in EIPs than "Main-net" (34 vs. 2 eips)

File: EIPS/eip-ice-age-fork.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+The primary problem with the Ice Age is that it is included in the complex mechanism that targets block times, which is an entirely separate in purpose. What is worse is due to being intwined with that algorithm, it is very difficult to simulate or predict its effect on the network. To predict the impact of the ice age, you must both make assumptions about the difficulty of main-net in the future, and predict the effect of changes in difficulty to the impact on the ice age and thus block-times.
+
+This fork will push back the Iceage as far as possible and will give us time to update the Iceage to no longer have these design problems. There are two solutions to consider with-in that time frame.
```
Comment (shemnon) [inline EIPS/eip-ice-age-fork.md]: 'as far as possible' -> 'as far as is reasonable.'  There is actually a floor of zero on the fake block period calculations which reference tests do test for, so we could have set the fake block at 30 million.

File: EIPS/eip-ice-age-fork.md (hunk 1 above)
Comment (MadeofTin) [reply EIPS/eip-ice-age-fork.md]: I wasn't sure if that block is affected or not. I wanted to pick the last possible block in that period.

File: EIPS/eip-ice-age-fork.md (hunk 4 above)
Comment (MadeofTin) [reply EIPS/eip-ice-age-fork.md]: updated

File: EIPS/eip-ice-age-fork.md (hunk 3 above)
Comment (MadeofTin) [reply EIPS/eip-ice-age-fork.md]: updated

File: EIPS/eip-2387.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+---
+eip: 2387
+title: "Hardfork Meta: Mountain Glacier"
```
Comment (axic) [inline EIPS/eip-2387.md]: Suggest "Muir Glacier" ([see here](https://en.wikipedia.org/wiki/Muir_Glacier)) which has retreated significantly. Or any other similar glacier :wink:

File: EIPS/eip-2387.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+author: James Hancock (@madeoftin)
+discussions-to: https://github.com/ethereum/EIPs/pull/2387
+type: Meta
+status: Draft
+created: 2019/11/22
```
Comment (axic) [inline EIPS/eip-2387.md]: Wrong date format.

File: EIPS/eip-2387.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
+---
+eip: 2387
+title: "Hardfork Meta: Mountain Glacier"
+author: James Hancock (@madeoftin)
+discussions-to: https://github.com/ethereum/EIPs/pull/2387
```
Comment (axic) [inline EIPS/eip-2387.md]: Discussion URL cannot be a pull request according to EIP-1.

File: EIPS/eip-2387.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+- Codename: Mountain Glacier
+
+### Activation
+  - `Block >= 9,200,000` on the Ethereum mainnet
+  - `Block >= tbd` on the Ropsten testnet
+  - `Block >= tbd` on the Kovan testnet
+  - `Block >= tbd` on the Rinkeby testnet
+  - `Block >= tbd` on the Görli testnet
+
+### Included EIPs EIP-2384
```
Comment (axic) [inline EIPS/eip-2387.md]: There seems to be a stray EIP-2384 text here.

File: EIPS/eip-2387.md (hunk 5 above)
Comment (econoar) [reply EIPS/eip-2387.md]: Or perhaps a characters name from the movie Ice Age? 😂

File: EIPS/eip-2387.md (hunk 7 above)
Comment (MadeofTin) [reply EIPS/eip-2387.md]: just making an Ethereum Magicans Thread

File: EIPS/eip-2387.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
+---
+eip: 2387
+title: "Hardfork Meta: Mountain Glacier"
+author: James Hancock (@madeoftin)
+discussions-to: https://ethereum-magicians.org/t/hard-fork-to-address-the-ice-age-eip-2387
+type: Meta
+status: Draft
+created: 2019-11-22
+requires: 1679
```
Comment (axic) [inline EIPS/eip-2387.md]: Should include 2384 here.

File: EIPS/eip-2387.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+### Activation
+  - `Block >= 9,200,000` on the Ethereum mainnet
+  - `Block >= tbd` on the Ropsten testnet
+  - `Block >= tbd` on the Kovan testnet
+  - `Block >= tbd` on the Rinkeby testnet
+  - `Block >= tbd` on the Görli testnet
+
+### Included EIPs
+  - [EIP-2384](https://github.com/ethereum/EIPs/pull/2384): Istanbul/Berlin Difficulty Bomb Delay
```
Comment (axic) [inline EIPS/eip-2387.md]: Please use the canonical EIP URL.

Comment (MadeofTin) [reply EIPS/eip-2387.md]: Earlier when I checked it wasn't there. Updating now.

File: EIPS/eip-2387.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+ - Update the mechanism so that behavior is predictable.
+ - Remove the Iceage entirely
+
+## Specification
+
+- Codename: Mountain Glacier
+
+### Activation
+  - `Block >= 9,200,000` on the Ethereum mainnet
```
Comment (holiman) [inline EIPS/eip-2387.md]: Should be `==`, not `>=`, right?

**Pull request merged**